### PR TITLE
Document MTLSPolicyConflict analyzer messages

### PR DESCRIPTION
**Pull request by selmanj**
Also included is some background on how configuration resources are matched (which is pretty much required to understand why a conflict occurs).

### CODE REVIEW

**Comment by selmanj** [comment]
Tagging @incfly, who is most qualified to vet what I've written for accuracy.

File: content/en/docs/reference/config/analysis/ist0113/index.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+To effectively resolve mutual TLS conflicts, it's helpful to understand how both
+destination rules and policies affect traffic to a service. Consider an example
+service named `my-service` in namespace `my-namespace`. To determine which
+policy object is applied to `my-service`, the following resources are matched in
+order:
+
+1. A policy resource in namespace `my-namespace` that contains a `target`
+   specifying `my-service`.
+1. A policy resource in namespace `my-namespace` that does not contain a
```
Comment (incfly) [inline content/en/docs/reference/config/analysis/ist0113/index.md]: fyi namespace has to be named `default`
https://github.com/istio/api/blob/master/authentication/v1alpha1/policy.proto#L271

File: content/en/docs/reference/config/analysis/ist0113/index.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+   `target`. This policy implicitly applies to the entire namespace.
```
Comment (incfly) [inline content/en/docs/reference/config/analysis/ist0113/index.md]: this ns scope is intended, not sure why we call it implicit?

**Comment by frankbu** [comment]
@selmanj Are you planning to make any changes based on the review comments?

File: content/en/docs/reference/config/analysis/ist0113/index.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+title: MTLSPolicyConflict
+layout: analysis-message
+---
+
+This message occurs when a destination rule resource and a policy resource are
+in conflict with regards to mutual TLS. The two resources are in conflict if they each
+specify incompatible mutual TLS modes to be used; this conflict means traffic matching
+the destination rule to the specified host will be rejected.
+
+This message is only produced on clusters that are not using [automatic mutual
```
Comment (sushicw) [inline content/en/docs/reference/config/analysis/ist0113/index.md]: should this be "service meshes" instead of "clusters"?

File: content/en/docs/reference/config/analysis/ist0113/index.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+{{< /text >}}
+
+Contained in this message are the two resources that are in conflict:
+
+* Policy resource `my-namespace/my-policy`, which is specifying 'Plaintext' as its
+  supported mutual TLS mode.
+* Destination rule resource `istio-system/default-rule`, which is requiring mutual TLS
+  when sending traffic to host `myhost.my-namespace.svc.cluster.local:8080`
+
+You can fix the conflict by:
```
Comment (sushicw) [inline content/en/docs/reference/config/analysis/ist0113/index.md]: Maybe rephrase to make it clear that the user needs to do one of the following, not all of the following.

e.g. "You can fix the conflict in several ways:"

 It's a bit unclear which 
 add "doing one of the following" or "doing all of the following" to the end of this?

File: content/en/docs/reference/config/analysis/ist0113/index.md (hunk 1 above)
Comment (selmanj) [reply content/en/docs/reference/config/analysis/ist0113/index.md]: Done.

File: content/en/docs/reference/config/analysis/ist0113/index.md (hunk 2 above)
Comment (selmanj) [reply content/en/docs/reference/config/analysis/ist0113/index.md]: I don't mean that its unintended by saying that its implicit. I mean it's implicit because the namespace itself is not listed as a target the same way that a service does (by listing no targets and naming it default, the target is the namespace even though it's not explicitly written).

File: content/en/docs/reference/config/analysis/ist0113/index.md (hunk 4 above)
Comment (selmanj) [reply content/en/docs/reference/config/analysis/ist0113/index.md]: Added "doing one of the following". Thanks!

File: content/en/docs/reference/config/analysis/ist0113/index.md (hunk 3 above)
Comment (selmanj) [reply content/en/docs/reference/config/analysis/ist0113/index.md]: Done.

File: content/en/docs/reference/config/analysis/ist0113/index.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+1. A mesh policy resource named `default`.
+
+To determine which destination rules are applied to traffic sent to
+`my-service`, we must first know which namespace the traffic originates from.
+For the sake of this example, let's call that namespace `other-namespace`.
+Destination rules are matched in the following order:
+
+1. A destination rule in namespace `other-namespace` that specifies a host that
+   matches `my-service.my-namespace.svc.cluster.local`. This may be an exact
+   match or a wildcard match. Also note that the `exportTo` field, which
```
[the remark this answers is not in the document]
Comment (selmanj) [reply content/en/docs/reference/config/analysis/ist0113/index.md]: I wanted to leave this out too, but include a note (e.g. if you're using values for exportTo, please refer to this page). I couldn't find a page to link to however that described the semantics of exportTo and how it's used in DR resolution

[the remark this answers is not in the document]
Comment (frankbu) [reply content/en/docs/reference/config/analysis/ist0113/index.md]: Sorry if I approved/merged this PR prematurely. I read the DR part and also concluded it was correct. I did add [this entry about DR resolution](https://istio.io/docs/ops/best-practices/traffic-management/#cross-namespace-configuration) to the op guide a little while ago, which I think is what you wanted, but now you will need to send another PR to change it, unfortunately. Btw, should this PR also get cherry-picked to release-1.4?
